catalog/core/catalog-core-localstorageprovider/src/main/java/org/codice/ddf/catalog/content/impl/FileSystemStorageProvider.java

```diff
@@ -455,9 +466,17 @@ private Path getTempContentItemDir(String requestId, URI contentUri) {
     private Path getContentItemDir(URI contentUri) {
         List<String> pathParts = getContentFilePathParts(contentUri.getSchemeSpecificPart(),
                 contentUri.getFragment());
-
-        return Paths.get(baseContentDirectory.toAbsolutePath()
-                .toString(), pathParts.toArray(new String[pathParts.size()]));
+        try {
+            return Paths.get(baseContentDirectory.toAbsolutePath()
```

Comment:
I think we want canonical, rather than absolute, paths here. Absolute paths can still contain symlinks, redirects, and OS expansion characters that may or may not be resolved correctly. [This](http://stackoverflow.com/a/1099370) is a good summary.